src/logjam/framework/timbre.clj

```diff
@@ -0,0 +1,94 @@
+(ns logjam.framework.timbre
+  "Log event capturing implementation for Timbre."
+  (:require
+   [logjam.appender :as appender]
+   [taoensso.timbre :as timbre])
+  (:import
+   (java.util Date)))
+
+(def ^:private log-levels
+  "The Timbre level descriptors."
+  (->> [:trace :debug :info :warn :error :fatal :report]
+       (map-indexed #(assoc {:name %2
+                             :category %2
+                             :object %2}
+                            :weight %1))))
+
+(defn- extract-event-data [{:keys [vargs level appender-id msg_ ^Date instant context ^Throwable ?err thread]}]
+  (cond-> {:arguments vargs ;; `vargs` are the arguments passed to a errorf, infof, etc call
+           :id (java.util.UUID/randomUUID)
+           :level level
+           :logger appender-id
+           :message (or (not-empty @msg_)
+                        (some-> ?err .getMessage))
+           :thread (or thread
+                       (.getName (Thread/currentThread)))
+           :timestamp (some-> instant .getTime)
+           :mdc (or context {})}
```

Comment:
Timbre contexts can be an arbitrary map, while MDCs are speced as String->String maps.

Maybe we could relax the `:logjam.event/mdc` Spec?

Reply:
Sounds good to me!